placement/algo/mirrored.go

```diff
@@ -285,6 +267,83 @@ func allInstancesInState(
 
 	return len(instanceIDs) == 0
 }
+
+func (a mirroredAlgorithm) returnInitializingShards(
```

Comment:
Can you add some comments briefly explaining the algorithm?

Reply:
sure